provider/go.go

```diff
@@ -35,92 +39,136 @@ func NewGoProviderFromConfig(other map[string]interface{}) (IntProvider, error)
+	if err != nil {
+		return nil, err
+	}
+
 	p := &Go{
 		vm:     vm,
 		script: cc.Script,
+		in:     in,
+		out:    out,
 	}
 
 	return p, nil
 }
 
 // FloatGetter parses float from request
 func (p *Go) FloatGetter() func() (float64, error) {
-	return func() (res float64, err error) {
-		v, err := p.vm.Eval(p.script)
-		if err == nil {
-			if typ := reflect.TypeOf(res); v.CanConvert(typ) {
-				res = v.Convert(typ).Float()
-			} else {
-				err = fmt.Errorf("not a float: %v", v)
-			}
+	return func() (float64, error) {
+		if err := handleInTransformation(p); err != nil {
+			return 0, err
+		}
+
+		v, err := p.evaluate()
+		if err != nil {
+			return 0, err
 		}
 
-		return res, err
+		return p.convertToFloat(v)
 	}
 }
 
 // IntGetter parses int64 from request
 func (p *Go) IntGetter() func() (int64, error) {
-	return func() (res int64, err error) {
-		v, err := p.vm.Eval(p.script)
-		if err == nil {
-			if typ := reflect.TypeOf(res); v.CanConvert(typ) {
-				res = v.Convert(typ).Int()
-			} else {
-				err = fmt.Errorf("not an int: %v", v)
-			}
+	return func() (int64, error) {
+		if err := handleInTransformation(p); err != nil {
+			return 0, err
 		}
 
-		return res, err
+		v, err := p.evaluate()
+		if err != nil {
+			return 0, err
+		}
+
+		return p.convertToInt(v)
 	}
 }
 
 // StringGetter parses string from request
 func (p *Go) StringGetter() func() (string, error) {
-	return func() (res string, err error) {
-		v, err := p.vm.Eval(p.script)
-		if err == nil {
-			if typ := reflect.TypeOf(res); v.CanConvert(typ) {
-				res = v.Convert(typ).String()
-			} else {
-				err = fmt.Errorf("not a string: %v", v)
-			}
+	return func() (string, error) {
+		if err := handleInTransformation(p); err != nil {
+			return "", err
 		}
 
-		return res, err
+		v, err := p.evaluate()
+		if err != nil {
+			return "", err
+		}
+
+		return p.convertToString(v)
 	}
 }
 
 // BoolGetter parses bool from request
 func (p *Go) BoolGetter() func() (bool, error) {
-	return func() (res bool, err error) {
-		v, err := p.vm.Eval(p.script)
-		if err == nil {
-			if typ := reflect.TypeOf(res); v.CanConvert(typ) {
-				res = v.Convert(typ).Bool()
-			} else {
-				err = fmt.Errorf("not a boolean: %v", v)
-			}
+	return func() (bool, error) {
+		if err := handleInTransformation(p); err != nil {
+			return false, err
+		}
+
+		v, err := p.evaluate()
+		if err != nil {
+			return false, err
 		}
 
-		return res, err
+		return p.convertToBool(v)
 	}
 }
 
 func (p *Go) paramAndEval(param string, val any) error {
+	err := p.setParam(param, val)
+	if err != nil {
+		return err
+	}
+
+	v, err := p.evaluate()
+	if err != nil {
+		return err
+	}
+
+	return handleOutTransformation(p, v)
+}
+
+func (p *Go) setParam(param string, val any) error {
+	if str, ok := val.(string); ok {
+		val = "\"" + str + "\""
+	}
+
 	_, err := p.vm.Eval(fmt.Sprintf("%s := %v;", param, val))
 	if err == nil {
-		_, err = p.vm.Eval(p.script)
+		_, err = p.vm.Eval(fmt.Sprintf("param := %v;", param))
```

Comment:
why do we need the param/val thing? We should only offer one way of how things work to avoid confusion.

Reply:
We don't need it, this sneaked in from my other PR. I will remove it in the go implementation. What about javascript? That implementation still sets param/val. I personally would like to get rid of it, but that might break existing configurations.

Reply:
Lets remove it.

Reply:
Removed and changed the demo.yaml accordingly